### DISCUSSION TITLE
Created new templates doesnt shown on visual studio

### DISCUSSION THREAD

**Discussion by OmarPathways** [dotnet/templating#8462] [Template authoring] [answered]
I created a new template based on an existing solution and installed it using dotnet new. Why can't I find this template in Visual Studio after installing it via the CLI?

I checked an online resource for guidance, https://devblogs.microsoft.com/dotnet/net-cli-templates-in-visual-studio/ but it seems outdated. Where can I find the option for installing templates in the Preview Features of Visual Studio 2022?

Are there any specific steps I need to follow to ensure my new template appears in Visual Studio?

Could there be compatibility issues between the template I created and Visual Studio 2022 that would prevent it from being recognized?

Is there a recommended way to troubleshoot template visibility issues in Visual Studio?

**Comment by sayedihashimi** [comment, marked as answer]
I have some updated guidance at https://aka.ms/netcore-templates.

If the template appears when you run `dotnet new list` it should also appear in Visual Studio. If that is not happening, if you can give some more details on what is happening, and a sample to reproduce that we can investigate the issue.

**Reply by OmarPathways** [reply]
the below steps in the shared link solved everything 
thanks alot 
![image](https://github.com/user-attachments/assets/df13885e-b425-4644-be22-0b661a5885a4)
